render/render-module.nix

```
@@ -587,4 +596,63 @@ in
       };
     }
   );
+  config.flake.modules.flakePartsRenderInput = {
+    github =
+      inputArgs:
+      let
+        cfg = inputArgs.config;
+      in
+      {
+        options = {
+          owner = lib.mkOption {
+            type = lib.types.str;
+          };
+          repo = lib.mkOption {
+            type = lib.types.str;
+          };
+        };
+        config = {
+          baseUrl = lib.mkIf (cfg.rev != null) (
+            lib.mkOverride 1400 "https://github.com/${cfg.owner}/${cfg.repo}/blob/${cfg.rev}${cfg.sourceSubpath or ""}"
+          );
+
+          flakeRef = lib.mkOptionDefault "github:${cfg.owner}/${cfg.repo}";
+        };
+      };
+
+    gitlab =
+      inputArgs:
+      let
+        cfg = inputArgs.config;
+      in
+      {
+        options = {
+          forgeRoot = lib.mkOption {
+            type = lib.types.str;
+            default = "https://gitlab.com/";
+          };
+          owner = lib.mkOption {
+            type = lib.types.str;
+          };
+          repo = lib.mkOption {
+            type = lib.types.str;
+          };
+        };
+        config = {
+          baseUrl = lib.mkIf (cfg.rev != null) (
+            lib.mkOverride 1400 "${cfg.forgeRoot}${cfg.owner}/${cfg.repo}/-/blob/${cfg.rev}"
+          );
+
+          flakeRef = lib.mkOptionDefault (
+            let
+              path_ = "${cfg.owner}/${cfg.repo}";
+            in
+            if cfg.forgeRoot == inputArgs.options.forgeRoot.default then
+              "gitlab:${path_}"
+            else
+              "git+${cfg.forgeRoot}${path_}"
```

Example: `gitlab:openldap/openldap?host=git.openldap.org`

```suggestion
              "gitlab:${path_}?host=${cfg.forgeRoot}"
```